ivadomed/config/config_pose.json

```diff
@@ -0,0 +1,94 @@
+{
+    "command": "train",
+    "gpu_ids": [0],
+    "path_output": "data/example/labeling_t0test1",
+    "model_name": "pose_model",
+    "debugging": false,
+    "loader_parameters": {
+        "path_data": ["data/data-multi-subject"],
+        "target_suffix": ["_heatmap0"],
+        "extensions": [".nii.gz"],
+        "roi_params": {
+            "suffix": null,
+            "slice_filter_roi": null
+        },
+        "contrast_params": {
+            "training_validation": ["T1w"],
+            "testing": [ "T1w"],
+            "balance": {}
+        },
+        "slice_filter_params": {
+            "filter_empty_mask": true,
+            "filter_empty_input": true
+        },
+        "slice_axis": "sagittal",
+        "multichannel": false,
+	    "soft_gt": true
+    },
+    "split_dataset": {
+        "fname_split": null,
+        "random_seed": 8,
+        "split_method" : "participant_id",
+        "data_testing": {"data_type": null, "data_value":[]},
+        "balance": null,
+        "train_fraction": 0.6,
+        "test_fraction": 0.2
```

Comment:
I've run `--test` using the T1w model (mostly because I'm trying to learn how to use `ivadomed`, and I was curious what the output was). Here are the metrics for the 52-subject test dataset: [`evaluation_3Dmetrics.csv`](https://github.com/ivadomed/ivadomed/files/7164215/evaluation_3Dmetrics.csv)

Skimming the results, however, I notice that ivadomed is calculating metrics as though this were a segmentation problem. For example, take the subject `sub-balgrist05`:

![Screenshot from 2021-09-14 14-15-38](https://user-images.githubusercontent.com/16181459/133311733-37126197-37f1-4f72-889b-c45c63231414.png)

Here are the resulting metrics for this subject:

| image_id                   | avd_class0        | dice_class0       | lfdr_0-20vox_class0 | lfdr_21-100vox_class0 | lfdr_class0 | ltpr_101-INFvox_class0 | ltpr_class0 | mse_class0        | n_101-INFvox | n_gt_class0 | n_pred_class0 | precision_class0  | recall_class0    | rvd_class0        | specificity_class0 | vol_gt_class0    | vol_pred_class0 | lfdr_101-INFvox_class0 |
|----------------------------|-------------------|-------------------|---------------------|-----------------------|-------------|------------------------|-------------|-------------------|--------------|-------------|---------------|-------------------|------------------|-------------------|--------------------|------------------|-----------------|------------------------|
| sub-balgrist05_rec-mid_T1w | 0.418087197212938 | 0.537764707940283 |                     | 1                     | 0.5         | 1                      | 1           | 0.564196995428872 | 1            | 1           | 6             | 0.730948754094583 | 0.42534843818889 | 0.418087197212938 | 0.999298371317197  | 371.189633507755 | 216             |                        |

Since SCT is more concerned with single-voxel labels, rather than disc heatmaps, the metrics above don't exactly tell the whole picture.

Reply:
I notice that in the [`vertebral-labeling-validation`](https://github.com/sct-pipeline/vertebral-labeling-validation/blob/800d9db0810a9cba53b9d85a157366e582c1bf48/run_prediction_deep.sh#L59-L60) repository, @lrouhier chose to look for [missing discs](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/67170203934274498b90bc65f6a084f724b6ec19/spinalcordtoolbox/labels.py#L278-L304) and [MSE](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/67170203934274498b90bc65f6a084f724b6ec19/spinalcordtoolbox/labels.py#L308-L333) instead. I will keep this in mind when evaluating the performance of the various methods.